stdlib/datetime.pyi

```diff
@@ -10,6 +11,7 @@ elif sys.version_info >= (3, 9):
     __all__ = ("date", "datetime", "time", "timedelta", "timezone", "tzinfo", "MINYEAR", "MAXYEAR")
 
 _D = TypeVar("_D", bound=date)
+_IntLike: TypeAlias = int | SupportsIndex
```

Comment:
I don't think this is necessary -- `int`s have an `__index__` method:

```pycon
>>> x = 3
>>> x.__index__()
3
```

You can just use `SupportsIndex`, and it will allow `int`s

Reply:
> I kept it for extra readability.

To me it makes it less readable, as it causes confusion -- I'm left wondering why there's a redundant union here, since I know that `int` is a subtype of `SupportsIndex`.

> `int` is the main data-type, `SupportsIndex` is the second option.

Sure, but that applies in 99% of cases where the runtime accepts anything with an `__index__` method, and we don't use a special type alias for any of those:

https://github.com/python/typeshed/blob/b6564d976b336ddff962870092ac873a60dfa9b2/stdlib/builtins.pyi#L291

https://github.com/python/typeshed/blob/b6564d976b336ddff962870092ac873a60dfa9b2/stdlib/builtins.pyi#L1171

https://github.com/python/typeshed/blob/b6564d976b336ddff962870092ac873a60dfa9b2/stdlib/builtins.pyi#L993-L1001